lib/phoenix/live_dashboard/system_info.ex

```diff
@@ -235,10 +244,103 @@ defmodule Phoenix.LiveDashboard.SystemInfo do
     end
   end
 
+  ## Socket callbacks
+
+  def sockets_callback(search, sort_by, sort_dir, limit) do
+    sorter = if sort_dir == :asc, do: &<=/2, else: &>=/2
+
+    sockets =
+      for port <- Port.list(), info = socket_info(port), show_socket?(info, search), do: info
+
+    count = length(sockets)
+
+    sockets =
+      sockets
+      |> Enum.sort_by(&Keyword.fetch!(&1, sort_by), sorter)
+      |> Enum.take(limit)
+
+    {sockets, count}
+  end
+
+  def socket_info_callback(port, keys) do
+    case socket_info(port) do
+      :error -> :error
+      info -> Keyword.take(info, keys)
+    end
+  end
+
+  defp socket_info(port) do
+    with info when not is_nil(info) <- Port.info(port),
```

Comment:
You want to filter inet ports here, and not below, so you don't call inet.getstat for ports that are not sockets.

```suggestion
    with info when not is_nil(info) <- Port.info(port),
         true <- info[:name] in @inet_ports,
```